File: articles/problems-procedures-and-formalization/cpp/formalization_audit.cpp
#include <iostream>
#include <string>
#include <vector>

struct FormalizationCase {
  std::string name;
  double input;
  double output;
  double objective;
  double assumptions;
  double governance;
};

int main() {
  std::vector<FormalizationCase> cases = {
    {"Document search", 82, 78, 70, 58, 56},
    {"Worker scheduling", 72, 76, 58, 54, 62},
    {"Public service triage", 60, 72, 52, 46, 66},
    {"Scientific simulation", 86, 80, 76, 84, 70}
  };
  std::cout << "case_name,formalization_score,warning\n";
  for (const auto& c : cases) {
    double score = 0.20*c.input + 0.20*c.output + 0.25*c.objective + 0.20*c.assumptions + 0.15*c.governance;
    std::cout << c.name << "," << score << ",Synthetic educational diagnostic only.\n";
  }
}

File: articles/problems-procedures-and-formalization/typescript/src/formalizationAudit.ts
export type FormalizationCase = {
  name: string;
  input: number;
  output: number;
  objective: number;
  assumptions: number;
  governance: number;
};

export function formalizationScore(item: FormalizationCase): number {
  return 0.20 * item.input + 0.20 * item.output + 0.25 * item.objective + 0.20 * item.assumptions + 0.15 * item.governance;
}
